Add GitHub workflows for Windows

stefanb2 opened the pull request:
Untested.

I hope that after #36 has been merged that then workflow changes can be tested by pushing a PR.

stefanb2 committed:
build: add GitHub workflows for Windows

- matrix: build against Qt 5 & 6

stefanb2 commented:
Ok, as expected: after merging the initial github workflow PR it is now possible to test CI changes with further PRs.

stefanb2 commented:
@bkueng looks like Windows build no longer compiles and requires source code fixes. I'll leave this in draft for now.

bkueng commented:
I can have a look myself. We can merge it as is if you like, and I'll do the follow-up.

bkueng committed:
fix keypress_windows: need to use scope for enums

bkueng committed:
fix keypress_windows: use LPCWSTR (const wchar_t*) instead of LPCTSTR (const char*)

fixes this error:
D:\a\qMasterPassword\qMasterPassword\src\keypress_windows.cpp(99,5): error C2664: 'void CKeyboardSimulator::TextEntry(LPCTSTR)': cannot convert argument 1 from 'const _Elem *' to 'LPCTSTR' [D:\a\_temp\build\qMasterPassword.vcxproj]

bkueng committed:
CMakeLists: use find_package(OpenSSL REQUIRED) for OpenSSL

bkueng committed:
ci: add scrypt to windows build

bkueng committed:
refactor logging: convert ELOG enum to an enum class & rename

ERROR was conflicting with a macro on Windows.

bkueng committed:
fix: use _WIN32 instead of WIN32

_WIN32 is defined by the compiler whereas WIN32 by the SDK.

bkueng committed:
.gitignore: ignore cmake-build-*

stefanb2 committed:
build: use CXXFLAGS to add flags

Setting CMAKE_CXX_FLAGS on the command line overrides the defaults
generated by CMake, which leads to compiler warnings and broken code. We
want to add flags to the compiler command line, hence setting CXXFLAGS
environment variable is the correct approach.

- CMakeLists.txt: prepend linker options to address linker warnings
- fix some typos
- reformat run parts to improve readability
